llvm/test/CodeGen/RISCV/rvv/fixed-vectors-shufflevector-vnsrl.ll

```diff
@@ -822,7 +820,7 @@ define void @vnsrl_32_i32_two_source(ptr %in0, ptr %in1, ptr %out) {
 ; V-NEXT:    vle32.v v8, (a0)
 ; V-NEXT:    vle32.v v9, (a1)
 ; V-NEXT:    vmv.v.i v0, 1
-; V-NEXT:    vrgather.vi v9, v8, 1, v0.t
+; V-NEXT:    vslidedown.vi v9, v8, 1, v0.t
```

Comment:
A masked vrgather.vi is one cycle faster than a masked vslidedown.vi on the bpi-f3 according to https://camel-cdr.github.io/rvv-bench-results/bpi_f3/index.html. Is there an easy way to avoid this? Check that the src lengths > 2?

Reply:
I think this is a BP3 oddity, and not something we should tune for.  Having vrgather.vi, vslidedown.vi and vslideup.vi all be the same cost at mf2, but for there to be a constant 1 cycle difference at all higher LMUL seems quite odd.  I'm honestly a bit suspicious of the numbers at all.